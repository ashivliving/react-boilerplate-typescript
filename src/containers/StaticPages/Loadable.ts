/**
 * Asynchronously loads the component for StaticPages
 * Created by Bantai:sahil for Turtlemint on 2/26/2019, 7:01:16 PM
 */

import loadable from '@loadable/component';

export default loadable(() => import('./index'));
