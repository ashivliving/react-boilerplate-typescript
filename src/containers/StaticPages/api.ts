/**
 * StaticPages API Endpoints
 * Contains api calls and endpoints for StaticPages
 * Created by Bantai:sahil for Turtlemint on 2/26/2019, 7:01:16 PM
 */

import { Observable } from 'rxjs';
import { http } from '../../utils/http';

/**
 * Your main api call to outside this environment goes from here
 * Make sure your API returns an Observable of strict type of your contract that is defined in the interface
 */
export const staticPagesApiCall: () => Observable<string> =
    (): Observable<string> =>
        http.get<string>('http://www.mocky.io/v2/5c7915e43000007d3149b1c0');
