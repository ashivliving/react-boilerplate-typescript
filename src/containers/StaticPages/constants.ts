/**
 * StaticPages constants
 * Created by Bantai:sahil for Turtlemint on 2/26/2019, 7:01:16 PM
 */

import { createApiConstants, createConstants } from '../../utils/actioncreator';

/*
* This creates 2 constants respectively
* DEFAULT.ACTION = 'staticPages/OPTIONAL_PREFIX_action';
* DEFAULT.GO = 'staticPages/OPTIONAL_PREFIX_go';
* If you dont want to further group/namespace remove optional_prefix
* Check the file src/utils/actioncreator.ts for details
*/
export const DEFAULT = createConstants('staticPages', 'optional_prefix')('ACTION', 'GO');

/*
* This creates 3 constants respectively for EPIC calls only
* DEFAULT_API.REQUEST = 'staticPages/REQUEST';
* DEFAULT_API.SUCCESS = 'staticPages/SUCCESS';
* DEFAULT_API.FAILURE = 'staticPages/FAILURE';
*/
export const DEFAULT_API = createApiConstants('staticPages');
