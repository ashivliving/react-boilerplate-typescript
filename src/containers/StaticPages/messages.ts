/*
 * StaticPages Messages
 *
 * This contains all the text/messages for the StaticPages container.
 * Created by Bantai:sahil for Turtlemint on 2/26/2019, 7:01:16 PM
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.StaticPages';

export default defineMessages({
  clickme: {
    defaultMessage: 'Async Operation!',
    id: `${scope}.clickme`,
  },
  header: {
    defaultMessage: 'This is the StaticPages container!',
    id: `${scope}.header`,
  },
  idlabel: {
    defaultMessage: 'ID:',
    id: `${scope}.idlabel`,
  },
  styleText: {
    defaultMessage: 'This is the styled text!',
    id: `${scope}.styleText`,
  },
  titlelabel: {
    defaultMessage: 'Title:',
    id: `${scope}.titlelabel`,
  },
});
