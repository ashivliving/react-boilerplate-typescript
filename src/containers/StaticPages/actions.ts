/**
 * StaticPages actions
 * Action-Creators for StaticPages
 * Created by Bantai:sahil for Turtlemint on 2/26/2019, 7:01:16 PM
 */
import { createAction, createApiAction } from '../../utils/actioncreator';
import { DEFAULT, DEFAULT_API } from './constants';

/**
 * This creates 1 default action.
 * defaultAction = function (payload) { return { type: DEFAULT.ACTION, payload: payload}};
 * You can simply dispatch that action using dispatch(defaultAction());
 * The payload is optional and can be omitted if not required.
 */
export const defaultAction = createAction(DEFAULT.ACTION);

/**
 * This creates 3 default API actions.
 * defaultAction.request = function (payload) { return { type: DEFAULT_API.REQUEST}};
 * defaultAction.success = function (payload) { return { type: DEFAULT_API.SUCCESS, payload: payload}};
 * defaultAction.failure = function (error) { return { type: DEFAULT_API.REQUEST, payload: error}};
 * You can simply dispatch request action using dispatch(defaultAPIAction.request());
 * You can simply dispatch success action using dispatch(defaultAPIAction.success(payload));
 * You can simply dispatch failure action using dispatch(defaultAPIAction.failure(error));
 * The payload is optional and can be omitted if not required.
 */
export const defaultAPIAction = createApiAction(DEFAULT_API);
