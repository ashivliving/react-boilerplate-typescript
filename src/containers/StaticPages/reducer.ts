/*
 * StaticPages reducer
 * All reducers for StaticPages goes here
 * Created by Bantai:sahil for Turtlemint on 2/26/2019, 7:01:16 PM
 */

import { fromJS } from 'immutable';
import { createReducer } from '../../utils/actioncreator';
import { DEFAULT , DEFAULT_API } from './constants';
import { IStaticPagesApiContract } from './interface';

const initialState: IStaticPagesApiContract = fromJS({});

/**
 * All your reducers go here based on the constants that are created in the constants file
 */
const staticPagesReducer = createReducer(initialState, {
  [DEFAULT.ACTION]: (state: object) => ({...state, ...initialState}),
  [DEFAULT_API.SUCCESS]: (state: object, action: any) => ({...state, ...action}),
  [DEFAULT_API.FAILURE]: (state: object, action: any) => ({...state, ...action}),
});

export default staticPagesReducer;
