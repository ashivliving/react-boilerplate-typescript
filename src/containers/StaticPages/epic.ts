/**
 * staticPages Epics
 * Contains epics for StaticPages
 * Created by Bantai:sahil for Turtlemint on 2/26/2019, 7:01:16 PM
 */

import { Action } from 'redux';
import { ActionsObservable, ofType, StateObservable } from 'redux-observable';
import { Observable, of } from 'rxjs';
import { catchError, map, switchMap, tap } from 'rxjs/operators';
import { defaultAPIAction } from './actions';
import { staticPagesApiCall } from './api';
import { DEFAULT_API} from './constants';

export const staticPagesEpic = (action$: ActionsObservable<Action>, state$: StateObservable<Action>):
    Observable<string | object> => action$
    .pipe(
        ofType(DEFAULT_API.REQUEST),
        tap((action: Action) => {
            // console.log('Do your awesome Epic for Form in this area', action)
        }),
        switchMap(() => staticPagesApiCall()
            .pipe(
                map((payload) => defaultAPIAction.success(payload)),
                catchError((err) => of(defaultAPIAction.failure(err))),
            ),
        ),
    );

export default staticPagesEpic;
