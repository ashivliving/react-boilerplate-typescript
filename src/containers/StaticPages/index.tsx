/*
 * StaticPages
 * Created by Bantai:sahil for Turtlemint on 2/26/2019, 7:01:16 PM
 */

import React, {useCallback, useEffect, useState} from 'react';
import { Helmet } from 'react-helmet';
import { FormattedMessage } from 'react-intl';
import {useDispatch, useMappedState} from 'redux-react-hook';
import { addEpic } from '../../rootEpic';
import { useInjectReducer } from '../../utils/reducerInjectors';
import {defaultAPIAction} from './actions';
import epic from './epic';
import messages from './messages';
import reducer from './reducer';
import { Title } from './styles';

addEpic(epic);
useInjectReducer('staticPagesReducer', reducer);

const StaticPages = (props: object) => {
  const dispatch = useDispatch();
  useEffect(() => {
      dispatch(defaultAPIAction.request());
  }, [false]);
  const mapState = useCallback(
        (state) => state.get('staticPagesReducer'),
        [1],
    );
  const {payload} = useMappedState(mapState);
  return (
    <div>
        <Helmet title='StaticPages' meta={[{content: 'Description of StaticPages', name: 'description'}]}  />
        <Title><FormattedMessage {...messages.styleText} /></Title>
        <FormattedMessage {...messages.header} />
          <div dangerouslySetInnerHTML={{__html: payload}}/>
    </div>
  );
};

export default StaticPages;
