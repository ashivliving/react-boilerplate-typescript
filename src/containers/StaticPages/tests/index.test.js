import { render } from 'enzyme';
import React from 'react';

import StaticPages from '../Loadable';

describe('<StaticPages />', () => {
  it('Expect to have unit tests specified', () => {
    const component = render(<StaticPages debug />);

    expect(component).toMatchSnapshot();
  });
});

export {};
