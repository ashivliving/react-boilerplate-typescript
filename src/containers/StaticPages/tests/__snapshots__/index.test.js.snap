// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaticPages /> Expect to have unit tests specified 1`] = `null`;
